doc/how_tos/transform_plugins.rst

```diff
@@ -68,8 +68,7 @@ Module, Class, Function etc. In our case we need to transform a class. It can be
 
   import astroid
 
-  def register(linter):
-    # Needed for registering the plugin.
+  def register(linter: "PyLinter") -> None:
```

Comment:
Why do we remove the doc string? Doesn't it make sense to keep some (updated) form?

Reply:
I was thinking maybe the docstring in PyLinter was enough. What about:

```suggestion
  def register(linter: "PyLinter") -> None:
      """Register the plugin inside the PyLinter."""
```
? (Obviousely the same docstring for each function)

Reply:
Yeah, but when you're looking at a checker-file (as a new contributor) you're not necessarily looking at the `PyLinter` file as well. Especially since it is not a class method these docstring are not necessarily "linked".

What about "Register the plugin with a linter. Function called during initialisation to register checkers." Can't hurt to be as explicit as possible.

Reply:
We could even include the `param` line as well if we're going to make them uniform. I think `param` is used by `sphinx`.

Reply:
Done, it's a lot of duplication imo but this is an issue with a design we cannot change.